File: packages/backend/src/bot/commands/room.ts
import { database } from "../../database";
import { Command, mentionUser, type User } from "../core";

export class RoomCommand extends Command {
	protected async onExecute(user: User, _: string[]): Promise<string | null> {
		const room = await database.trackmaniaRoom.findUnique({
			where: {
				channelId: this.channelId,
			},
		});
		if (room === null) {
			return `${mentionUser(user.displayName)} I am currently not in a room.`;
		}

		const message = `${room.name} [${room.numberOfPlayers}/${room.maxPlayers}]`;
		return `${mentionUser(user.displayName)} ${message}`;
	}
}

File: wheelgpt-go/internal/bot/commands/room.go
package commands

import (
	"context"
	"fmt"

	"github.com/SoWieMarkus/wheelgpt/internal/bot"
	"github.com/SoWieMarkus/wheelgpt/internal/db"
)

type RoomCmd struct{ Store *db.Store }

func (c *RoomCmd) Names() []string    { return []string{"room"} }
func (c *RoomCmd) Level() bot.AccessLevel { return bot.AccessUser }

func (c *RoomCmd) Execute(ctx *bot.CmdContext) (string, error) {
	row, err := c.Store.GetTrackmaniaRoom(context.Background(), ctx.ChannelID)
	if err != nil || row.ChannelID == "" {
		return fmt.Sprintf("%s No room data available.", bot.MentionUser(ctx.User.DisplayName)), nil
	}
	return fmt.Sprintf("Current room: %s (%d/%d players)", row.Name, row.NumberOfPlayers, row.MaxPlayers), nil
}
